services/azure_ai_service.py

```diff
@@ -0,0 +1,94 @@
+from services.ai_service import AIService
+from PIL import Image
+import openai
+import requests
+import os
+import io
+
+# See .env.example for Azure configuration needed
+from azure.cognitiveservices.speech import SpeechSynthesizer, SpeechConfig, ResultReason, CancellationReason
+
+class AzureAIService(AIService):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self.speech_key = os.getenv("AZURE_SPEECH_SERVICE_KEY")
+        self.speech_region = os.getenv("AZURE_SPEECH_SERVICE_REGION")
+
+        self.speech_config = SpeechConfig(subscription=self.speech_key, region=self.speech_region)
+        # self.speech_config.speech_synthesis_voice_name='en-US-JennyMultilingualV2Neural'
+
+        self.speech_synthesizer = SpeechSynthesizer(speech_config=self.speech_config, audio_config=None)
+        
+        self.languages = {
+            "french": {"lang": "fr-FR", "voices": ["fr-FR-HenriNeural", "fr-FR-DeniseNeural"]},
+            "spanish": {"lang": "es-MX", "voices": ["es-MX-JorgeNeural", "es-MX-DaliaNeural"]}
+        }
+        
+        self.speakers = []
+
+    def run_tts(self, message):
+        sentence = message['translation']
+        language = message['translation_language']
+        sid = message['session_id']
+        if not sid in self.speakers:
+            self.speakers.append(sid)
+        
+        print("⌨️ running azure tts async")
+        voice = self.languages[language]['voices'][self.speakers.index(sid)]
```

Comment:
Suggest creating a variable for `self.languages[language]` up above for reuse (vs re-accessing twice here and below)